### PR TITLE
Update PrestaShop deps: Symfony3, PHP 5.6 and PHPUnit 5.7

### PR DESCRIPTION
**Pull request by mickaelandrieu**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Update PrestaShop deps: Symfony3, PHP 5.6 and PHPUnit 5.7
| Type?         | improvement
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | http://forge.prestashop.com/browse/BOGOSS-56
| How to test?  | Travis test suite + visit all pages already migrated to Symfony

sf3_develop

> Note 8/11 : capture update with new debug toolbar

#### Important guidelines

* Make sure [your local branch is up to date](https://help.github.com/articles/syncing-a-fork/) before commiting your changes!
* Your code MUST respect [our Coding Standards](http://doc.prestashop.com/display/PS16/Coding+Standards) (for code written in PHP, JavaScript, HTML/CSS/Smarty/Twig, SQL)!
* Your commit name MUST respect our [naming convention](http://doc.prestashop.com/display/PS16/How+to+write+a+commit+message)!

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/8405)

### CODE REVIEW

File: install-dev/init.php [hunk 1]
```diff
@@ -79,6 +79,8 @@
 require_once _PS_INSTALL_PATH_.'classes/exception.php';
 require_once _PS_INSTALL_PATH_.'classes/session.php';
 
+\Symfony\Component\Debug\Debug::enable();
```
Comment (mickaelandrieu) [inline install-dev/init.php]: to be removed

File: install-dev/missing_requirement.php [hunk 2]
```diff
@@ -105,15 +105,15 @@
           PrestaShop installation requires at least the <b>zip extension</b> to be enabled.
       </li>
     <?php endif; ?>
-    <?php if (PHP_VERSION_ID < 50400): ?>
+    <?php if (PHP_VERSION_ID < 50509): ?>
       <li>
           PrestaShop requires at least PHP 5.4 or newer versions.
```
Comment (mickaelandrieu) [inline install-dev/missing_requirement.php]:
```diff
- PrestaShop requires at least PHP 5.4 or newer versions.
+ PrestaShop requires at least PHP 5.6 or newer versions.
```

and put PHP_VERSION_ID value to 50600

File: phpunit.xml [hunk 3]
```diff
@@ -10,7 +10,7 @@
     processIsolation            = "true"
```
Comment (mickaelandrieu) [inline phpunit.xml]: to be deleted, already exists => see `sf-tests.xml`

File: tests/Integration/classes/CartGetOrderTotalTest.php [hunk 4]
```diff
@@ -58,7 +58,7 @@ public static function setUpBeforeClass()
```
Comment (mickaelandrieu) [inline tests/Integration/classes/CartGetOrderTotalTest.php]: @todo => refactor `DatabaseDump` class to fix the security warning of MySQL 5.7.

File: tests/Unit/Classes/PrestaShopAutoloadTest.php [hunk 5]
```diff
@@ -55,9 +55,10 @@ public function testLoad()
         $this->assertTrue(class_exists('RequestSql', false));
     }
 
-    public function testClassLoadedFromCoreDir()
+    public function testClassFromCoreDirShouldntBeLoaded()
```
Comment (mickaelandrieu) [inline tests/Unit/Classes/PrestaShopAutoloadTest.php:58]: See https://github.com/PrestaShop/PrestaShop/blob/develop/classes/PrestaShopAutoload.php#L283

File: tests/Unit/Core/Payment/Core_Payment_PaymentOptionTest.php [hunk 6]
```diff
@@ -24,9 +24,9 @@
  * International Registered Trademark & Property of PrestaShop SA
  */
 
-namespace PrestaShop\PrestaShop\Tests\Unit\Core\Payment;
+namespace Tests\Unit\Core\Payment;
 
-use PrestaShop\PrestaShop\Tests\TestCase\UnitTestCase;
+use Tests\TestCase\UnitTestCase;
 use PrestaShop\PrestaShop\Core\Payment\PaymentOption as PaymentOption;
```
Comment (mickaelandrieu) [inline tests/Unit/Core/Payment/Core_Payment_PaymentOptionTest.php]: totaly useless ... to be removed

**Comment by Quetzacoalt91** [comment]
I just had a look on the Travis logs. It's hidden in the whole log files, but the product page seems to be broken:

`localhost:80 127.0.0.1 - - [12/Oct/2017:20:21:07 +0000] "GET /admin-dev/index.php/product/form/8?_token=9K90RRSDZZICKtPlDBNJ27Lw7duw-zve4mE8eRKKEw8 HTTP/1.1" 500 27059 "http://localhost/admin-dev/index.php/product/catalog?_token=9K90RRSDZZICKtPlDBNJ27Lw7duw-zve4mE8eRKKEw8" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/60.0.3112.113 Safari/537.36"`
https://travis-ci.org/PrestaShop/PrestaShop/jobs/287232660#L3247

File: src/PrestaShopBundle/Service/Hook/HookEvent.php [hunk 7]
```diff
@@ -64,7 +64,7 @@ public function getHookParameters()
 
         $sfContainer = SymfonyContainer::getInstance();
         if (!is_null($sfContainer) && !is_null($sfContainer->get('request_stack')->getCurrentRequest())) {
-            $globalParameters['request'] = $sfContainer->get('request');
+            $globalParameters['request'] = $sfContainer->get('request_stack')->getCurrentRequest();
```
Comment (Quetzacoalt91) [inline src/PrestaShopBundle/Service/Hook/HookEvent.php:67]: Same change to apply in `src/Adapter/HookManager`, line 58

https://github.com/mickaelandrieu/PrestaShop/blob/113a589a670aa1af2d1e91b7c0a6ba2e0094526a/src/Adapter/HookManager.php#L58

Comment (Quetzacoalt91) [reply src/PrestaShopBundle/Service/Hook/HookEvent.php:67]: See https://github.com/mickaelandrieu/PrestaShop/pull/9

Comment (mickaelandrieu) [reply src/PrestaShopBundle/Service/Hook/HookEvent.php:67]: yes we need to apply this change too, `request` service is not available anymore

File: src/PrestaShopBundle/Event/Dispatcher/NullDispatcher.php [hunk 8]
```diff
@@ -58,4 +58,8 @@ public function removeListener($eventName, $listener)
     public function removeSubscriber(EventSubscriberInterface $subscriber)
     {
     }
+
+    public function getListenerPriority($eventName, $listener)
```
Comment (Shine-neko) [inline src/PrestaShopBundle/Event/Dispatcher/NullDispatcher.php:62]: ?

Comment (mickaelandrieu) [reply src/PrestaShopBundle/Event/Dispatcher/NullDispatcher.php:62]: you need to implement the interface which have one more method ;)

File: tests/Integration/classes/CartGetOrderTotalTest.php (hunk 4 above)
Comment (mickaelandrieu) [reply tests/Integration/classes/CartGetOrderTotalTest.php]: not needed anymore :+1:

File: tests/Selenium/wdio.conf.js [hunk 9]
```diff
@@ -34,6 +34,9 @@ exports.config = {
     //
     capabilities: [{
         browserName: require('./settings').browserName,
+        chromeOptions: {
+          args: ['--headless', '--disable-gpu', '--window-size=1600,1200'],
```
Comment (mickaelandrieu) [inline tests/Selenium/wdio.conf.js:38]: 20% of performance for free :+1:

**Comment by wlisik** [comment]
Closed SISIM fuuckers

**Comment by LittleBigDev** [comment]
:lgtm:

---

Reviewed 198 of 269 files at r1, 19 of 33 files at r2, 17 of 20 files at r3, 2 of 3 files at r4, 1 of 2 files at r5, 41 of 41 files at r6, 3 of 3 files at r7.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8405#-:-KyH250xrSSBt-3fNI9D:bnfp4nl)*

**Comment by LittleBigDev** [comment]
I'm not approving yet, because there is still a blocking comment from @Quetzacoalt91 (in Reviewable)
But LGTM :)

**Review by Quetzacoalt91 — DISMISSED** [review]

**Comment by Quetzacoalt91** [comment]
The change I requested is done.

**Review by LittleBigDev — DISMISSED** [review]

**Comment by mickaelandrieu** [comment]
@Quetzacoalt91 now 1.7.3.x branch is done, is there anything that block the merge of Symfony 3 migration ? We'll have ~6 months to fix any related issues :)

**Review by Quetzacoalt91 — APPROVED** [review]

**Comment by eternoendless** [comment]
:lgtm:

---

Reviewed 1 of 269 files at r1, 1 of 2 files at r5, 5 of 6 files at r8, 6 of 6 files at r9.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8405#-:-Kz4WFiOUxiIrUh9PJlk:bnfp4nl)*

**Review by eternoendless — APPROVED** [review]

**Comment by eternoendless** [comment]
Thank you @mickaelandrieu 🎉

File: install-dev/missing_requirement.php [hunk 10]
```diff
@@ -105,15 +105,15 @@
...
-          PrestaShop requires at least PHP 5.4 or newer versions.
+          PrestaShop requires at least PHP 5.6 or newer versions.
           <i>You can install PrestaShop 1.6 if you can't update your version of PHP.</i>
       </li>
     <?php endif; ?>
-        <?php if (!is_writable(__DIR__.DIRECTORY_SEPARATOR.'..'.DIRECTORY_SEPARATOR.'app'.DIRECTORY_SEPARATOR.'cache')): ?>
+        <?php if (!is_writable(__DIR__.DIRECTORY_SEPARATOR.'..'.DIRECTORY_SEPARATOR.'var'.DIRECTORY_SEPARATOR.'cache')): ?>
       <li>
-          PrestaShop installation needs to write critical files in the folder app/cache.
+          PrestaShop installation needs to write critical files in the folder app/var.
```
Comment (Quetzacoalt91) [inline install-dev/missing_requirement.php:116]: Wrong, needs to be fixed

**Pull request merged**